### DISCUSSION TITLE
Write Queue

### DISCUSSION THREAD

**Discussion by DiKey98** [bluenviron/mediamtx#3513] [Questions] [answered]
### Question

Hello! There is a parameter in the config - named "writeQueueSize". It defines the queue size for output packets. Is this queue common for all streaming types (rtsp, webrtc, ...) or does each type have its own queue?

**Comment by aler9** [comment, marked as answer]
Hello, each reader has its own output queue, regardless of the protocol it uses for reading (currently RTSP, WebRTC, RTMP, HLS, SRT). In this way, a laggy reader cannot influence other ones. Parameter `writeQueueSize` influences the size of this queue. It's useful to touch it only when there are warnings in logs about the fact that the write queue is full.

**Reply by DiKey98** [reply]
Thanks!

[a comment by an automated account is not shown]
